### ISSUE TITLE
Option to validate MX record associated with email domains

### ISSUE THREAD

**Issue by lpellegr** [RohanNagar/jmail#87]
What a nice library you made, congratulations! I was wondering, what about adding an option to check that email domains have a valid MX record: at least one DNS record with an IP address?

**Comment by RohanNagar** [comment]
Thanks for the suggestion! I will spend some time in the next week looking into this. It may be a useful feature to add.

**Comment by lpellegr** [comment]
If this can help, here is a very basic method relying on dnsjava library:

```java
public boolean hasMxRecord(final String email) {
    final String[] chunks = email.split("@");
    if (chunks.length < 2) {
        return false;
    }

    try {
        final Record[] records = new Lookup(chunks[1], Type.MX).run();
        return records != null && records.length > 0;
    } catch (TextParseException e) {
        return false;
    }
}
```

**Comment by RohanNagar** [comment]
Thanks! The `dnsjava` library looks nice. I will use it as a reference when implementing this as I don't want to bring in any dependencies to `jmail`.

**Comment by RohanNagar** [comment]
It looks like this will be harder than I realized to implement without using an additional library. I plan to look into this some more, but if you need this functionality in the meantime, you can add your own validation rule to your `EmailValidator` like this:

```java
EmailValidator validator = JMail.strictValidator()
    .withRule(email -> {
      try {
        Records[] records = new Lookup(email.domainWithoutComments(), Type.MX).run();
        return records != null && records.length > 0;
      } catch (Exception e) {
        return false;
      }
    });

boolean valid = validator.isValid("test@test.com");
```

**Comment by RohanNagar** [comment]
Hi @lpellegr I have added this option to the library, which is now available in version `1.5.0`.

You can use it as such:

```java
EmailValidator validator = JMail.validator().requireValidMXRecord();

boolean valid = validator.isValid("test@test.com");
```

Please let me know if you run into any issues with this feature!
